### PR TITLE
Add documention on using alternative JVM vendor buildpacks.

### PR DESCRIPTION
**Pull request by dmikusa-pivotal**
## Summary

Includes:

 - a full list of the currently supported alternative JVM buildpacks.
 - instructions for manipulating `pack` to make this work
 - some formatting lint resolutions
 - a couple minor typo fixes

## Use Cases

Resolves https://github.com/paketo-buildpacks/paketo-website/issues/93

Well, most of #93, it defers on two points as these are larger/more complicated issues:

- how to create a builder/meta buildpack that pulls in this JVM distribution.
- a tutorial around how to use the libjvm library to create a JVM distro buildpack that's not currently supported.

I would be in favor of opening separate issues for these requests.

## Checklist
* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [N/A] I have added an integration test, if necessary.
* [N/A] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.

### CODE REVIEW

**Review by kvedurmu — APPROVED** [review]
Thanks for contributing this! IMO, I think simply documenting what alternative JVMs exist is good enough for now and we could tackle other stuff as a part of a separate set of issues. I'd like @ekcasey to take a look as well before we merge.

File: content/docs/buildpacks/language-family-buildpacks/java.md [hunk 1]
```diff
@@ -146,14 +155,17 @@ pack build samples/java \
 The resulting application image will be identical to that built in the Building with Maven example.
 
 ## About the JVM
+
 The Java Buildpack uses the [BellSoft Liberica][liberica] implementations of the JRE and JDK. JVM installation is handled by the [BellSoft Liberica Buildpack][bp/bellsoft-liberica]. The JDK will be installed in the build container but only the JRE will be contributed to the application image.
 
-See the [homepage][[bp/bellsoft-liberica]] for the Bellsoft Liberica Buildpackfor a full set of configuration options.
+See the [homepage][bp/bellsoft-liberica] for the Bellsoft Liberica Buildpackfor a full set of configuration options.
```
Comment (ekcasey) [inline content/docs/buildpacks/language-family-buildpacks/java.md:161]:
```suggestion
See the [homepage][bp/bellsoft-liberica] for the Bellsoft Liberica Buildpack for a full set of configuration options.
```
this was my typo from before but we might as well fix it while we are looking at it :)

File: content/docs/buildpacks/language-family-buildpacks/java.md [hunk 2]
```diff
@@ -164,39 +176,79 @@ pack inspect-image samples/app --bom | jq '.local[] | select(.name=="jre") | .me
...
+
+1. Buildpack-provided runtime components including the Memory Calculator accept semantically named environment variables which are then used to derive `JAVA_TOOL_OPTIONS` flags. Examples include:
     * `BPL_JVM_HEAD_ROOM`
     * `BPL_JVM_LOADED_CLASS_COUNT`
     * `BPL_JVM_THREAD_COUNT`
 2. Flags can be set directly at runtime with the `JAVA_TOOL_OPTIONS` environment variable. User-provided flags will be appended to buildpack-provided flags. If the user and a buildpack set the same flag, user-provided flags take precedence.
 
 See the [homepage][bp/bellsoft-liberica] for the Bellsoft Liberica Buildpack for a full set of configuration options.
 
+### Using Alternative JVM Vendors
```
Comment (ekcasey) [inline content/docs/buildpacks/language-family-buildpacks/java.md:207]:
```suggestion
### Using Alternative JVMs
```

File: content/docs/buildpacks/language-family-buildpacks/java.md [hunk 3]
```diff
@@ -164,39 +176,79 @@ pack inspect-image samples/app --bom | jq '.local[] | select(.name=="jre") | .me
...
+
+By default, the [Paketo Java buildpack][bp/java] will use the Liberica JVM. The following alternative JVM vendors may be substituted in place of Liberica's JVM.
```
Comment (ekcasey) [inline content/docs/buildpacks/language-family-buildpacks/java.md:209]:
```suggestion
By default, the [Paketo Java buildpack][bp/java] will use the Liberica JVM. The following Paketo JVM buildpacks may be used to substitute alternate JVM implemenations in place of Liberica's JVM.
```

Comment (ekcasey) [reply content/docs/buildpacks/language-family-buildpacks/java.md:209]: nit: I got a bit thrown off by the way we are using the word vendor. I know it was used in the title of the associated issue but if we are going to standardize on terminology I would rather call this category of buildpacks "JVM buildpacks" instead of "JVM vendor buildpacks".

File: content/docs/buildpacks/language-family-buildpacks/java.md [hunk 4]
```diff
@@ -164,39 +176,79 @@ pack inspect-image samples/app --bom | jq '.local[] | select(.name=="jre") | .me
...
+
+* [Paketo Amazon Corretto Buildpack][bp/amazon-corretto]
+* [Paketo Azul Zulu Buildpack][bp/azul-zulu]
+* [Paketo BellSoft Liberica Buildpack - Default][bp/bellsoft-liberica]
+* [Paketo Eclipse OpenJ9 Buildpack][bp/eclipse-openj9]
+* [Paketo GraalVM Buildpack][bp/graalvm]
+* [Paketo SapMachine Buildpack][bp/sap-machine]
```
Comment (ekcasey) [inline content/docs/buildpacks/language-family-buildpacks/java.md:216]: Maybe we should make this a two column table that associates JVM implementations with their respective buildpacks? I suggest this mostly because I think it would be nice to include links to the homepages of the JVMs themselves so folks can click through and read about the options.

File: content/docs/buildpacks/language-family-buildpacks/java.md [hunk 5]
```diff
@@ -164,39 +176,79 @@ pack inspect-image samples/app --bom | jq '.local[] | select(.name=="jre") | .me
...
+
+To use an alternative JVM vendor, you will need to set two `--buildpack` arguments to `pack build`, one for the alternative JVM vendor you'd like to use and one for the Paketo Java buildpack (in that order). This works because while you end up with two JVM vendor buildpacks, the first one, the one you're specifying will claim the buildplan entries so the second one will end up being a noop and doing nothing.
```
Comment (ekcasey) [inline content/docs/buildpacks/language-family-buildpacks/java.md:218]:
```suggestion
To use an alternative JVM, you will need to set two `--buildpack` arguments to `pack build`, one for the alternative JVM buildpack you'd like to use and one for the Paketo Java buildpack (in that order). This works because while you end up with two JVM buildpacks, the first one, the one you're specifying will claim the build plan entries so the second one will end up being a noop and doing nothing.
```

File: content/docs/buildpacks/language-family-buildpacks/java.md [hunk 6]
```diff
@@ -164,39 +176,79 @@ pack inspect-image samples/app --bom | jq '.local[] | select(.name=="jre") | .me
...
+
+This example will switch in the Azul Zulu buildpack:
+
+{{< code/copyable >}}
+pack build samples/jar --buildpack gcr.io/paketo-buidpacks/azul-zulu --buildpack paketo-buildpacks/java`
+{{< /code/copyable >}}
+
+There is one drawback to this approach. When using the method above to specify an alternative JVM vendor buildpack, this alternate buildpack ends up running before the CA certs buildpack and therefore traffic from the alternate JVM vendor buildpack won’t trust any additional CA certs. This is not expected to impact many users because JVM buildpacks should reach out to URLs that have a cert signed by a known authority with a CA in the default system truststore.
+
+If you have customized your JVM vendor buildpack to download the JVM from a URL that uses a certificate not signed by a well-known CA, you can workaround this by specifying the CA certs buildpack to run first. This works because while you will end up with the CA certificates buildpack specified twice, the lifecycle is smart enough to drop the second one.
```
Comment (ekcasey) [inline content/docs/buildpacks/language-family-buildpacks/java.md:228]:
```suggestion
If you have customized your JVM buildpack to download the JVM from a URL that uses a certificate not signed by a well-known CA, you can workaround this by specifying the CA certs buildpack to run first. This works because while you will end up with the CA certificates buildpack specified twice, the lifecycle is smart enough to drop the second one.
```

File: content/docs/buildpacks/language-family-buildpacks/java.md (hunk 3 above)
Comment (dmikusa-pivotal) [reply content/docs/buildpacks/language-family-buildpacks/java.md:209]: I have seen the term vendor used for different variants of OpenJDK, but I'm fine with keeping it short. I'm sometimes too wordy.

File: content/docs/buildpacks/language-family-buildpacks/java.md (hunk 4 above)
Comment (ekcasey) [reply content/docs/buildpacks/language-family-buildpacks/java.md]: looks great!

**Review by ekcasey — APPROVED** [review]

**Review by kvedurmu — APPROVED** [review]
